### PR TITLE
feat(ddm): Add url filter persisting

### LINKED ISSUE #56712: Store filters in the URL so that views are share-able

### PR DESCRIPTION
**Pull request by matejminar**
This PR adds a URL filter persisting. It means that mri, op, groupBy, query, and hiddenSeries are no longer stored in the local state but in the URL. This will enable persisting the view after a full page refresh and everything will be share-able just by copy-pasting the link.

Closes https://github.com/getsentry/sentry/issues/56712

### CODE REVIEW

**Review by iambriccardo — COMMENTED** [review]
I lack some knowledge to approve, but the logic looks good to me.

File: static/app/views/ddm/metricsExplorer.tsx [hunk 1]
```diff
@@ -178,19 +120,42 @@ function QueryBuilder({setQuery}: QueryBuilderProps) {
...
+                : availableOps[0];
+              router.push({
+                ...router.location,
+                query: {
+                  ...router.location.query,
+                  mri: option.value,
+                  op: selectedOp,
+                  groupBy: undefined,
+                },
+              });
```
Comment (ArthurKnaus) [inline static/app/views/ddm/metricsExplorer.tsx:137]: nit: this pattern is repeated quite often in this component. Does it make sense to create a helper for that?
```ts
const updateQuery = (partialQuery: Record<string, any>) => {
  router.push({
    ...router.location,
    query: {
      ...router.location.query,
      ...partialQuery,
    },
  })
}
```

**Review by ArthurKnaus — APPROVED** [review]
Nice 👍 
General nit: Maybe some multiline prop values can be moved out of the JSX into memoized variables. It is better for performance – but more important the JSX is cleaner and easier to read.

File: static/app/views/ddm/metricsExplorer.tsx (hunk 1 above)
Comment (matejminar) [reply static/app/views/ddm/metricsExplorer.tsx]: absolutely! done ❤️

**Pull request merged**